conda/recipes/cudf/meta.yaml

```
@@ -24,7 +24,7 @@ requirements:
     - setuptools
     - numba >=0.49.0
     - dlpack
-    - pyarrow 0.15.0.*
+    - pyarrow
```

Separately from `arrow-cpp` we depend on specifically `pyarrow 0.17.1` due to Cython linking and whatnot where we should probably just version this as such. This will also help in solves to prevent getting older libcudf nightlies where the constraints were looser on the arrow version.
```suggestion
    - pyarrow 0.17.1.*
```